### PR TITLE
Homework 1

### PR DESCRIPTION
**Pull request by Anujagagangras**
Initial code is committed to 

1. Used Pandas to import Dataframe
2. Used plotly to plot different plots.
3. Used sklearn standard scaler for standardizing the data.

### CODE REVIEW

File: Homework_1.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# importing libraries
+
+import pandas as pd
+import plotly.express as px
+from sklearn.preprocessing import StandardScaler
+
+data = pd.read_csv("/home/osboxes/Downloads/iris.csv")
```
Comment (cmalik3595) [inline Homework_1.py]: This is hardcoded path specific to your system environment. It will not work on other PCs

File: Homework_1.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+df1 = pd.DataFrame(data)
+df = pd.DataFrame(data)
+df.pop("class")
```
Comment (cmalik3595) [inline Homework_1.py]: Iris.data file doesn't contain any column names. Hence this statement is more of hard-coding.
Did you manually add column names to the Iris.data file ?

File: Homework_1.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+mean_value = n_array.mean(axis=0)
+min_value = n_array.min(axis=0)
+max_value = n_array.max(axis=0)
+
+print("mean = ", mean_value)
+print("min = ", min_value)
+print("max = ", max_value)
+
+
+Histogram = px.histogram(df, x="petalLength")
```
Comment (cmalik3595) [inline Homework_1.py]: Please add graphs showing the comparison of 
1. Different columns
2. Different Class of data

File: Homework_1.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+Histogram.show()
+
+scatter_plot = px.scatter(df, x="sepalLength", y="sepalWidth")
+scatter_plot.show()
+
+violin_plot = px.violin(df1, x="sepalLength", color="class")
```
Comment (cmalik3595) [inline Homework_1.py]: As per assignment, please add at least 5 different types of graphs

File: Homework_1.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+x = df.iloc[:, [0, 1, 2, 3]]
+x.head()
+
+scaler = StandardScaler().fit(x)
+print(scaler)
+
+x_scaled = scaler.transform(x)
+print(x_scaled)
+
```
Comment (cmalik3595) [inline Homework_1.py:64]: Please add remaining items
1. Random Forrest Classification
2. Pipelining with a combination of multiple classifier techniques
3. Unit test
4. NoseTest
5. Compare the Random forest and pipeline results

**Review by cmalik3595 — CHANGES_REQUESTED** [review]

File: Homework_1.py (hunk 2 above)
Comment (Anujagagangras) [reply Homework_1.py]: Yes. I manually added.

File: Homework_1.py (hunk 1 above)
Comment (Anujagagangras) [reply Homework_1.py]: Resolved this issue by importing dataset from sklearn.

File: Homework_1.py (hunk 3 above)
Comment (Anujagagangras) [reply Homework_1.py]: added graphs.

File: Homework_1.py (hunk 5 above)
Comment (Anujagagangras) [reply Homework_1.py:64]: I added random forest.

File: Homework_1.py [hunk 6]
```diff
@@ -2,42 +2,109 @@
...
-mean_value = n_array.mean(axis=0)
-min_value = n_array.min(axis=0)
-max_value = n_array.max(axis=0)
 
-print("mean = ", mean_value)
-print("min = ", min_value)
-print("max = ", max_value)
+# 2. getting sum simple summary statistics (mean, min, max) using numpy
+def numpy_statistics():
+    n_array = iris.to_numpy()
```
Comment (cmalik3595) [inline Homework_1.py:40]: iris is a local variable in this def. Please add global keyword to include the global variable "iris"

File: Homework_1.py [hunk 7]
```diff
@@ -2,42 +2,109 @@
...
-scatter_plot.show()
+# 3. Different types of plots (Histogram, scatter plot, violin, boxplot, barchart)
 
-violin_plot = px.violin(df1, x="sepalLength", color="class")
-violin_plot.show()
 
-x = df.iloc[:, [0, 1, 2, 3]]
-x.head()
+def plot_graphs():
+    histogram = px.histogram(iris, x="petal length")
```
Comment (cmalik3595) [inline Homework_1.py:56]: iris is a local variable in this def. Please add global keyword to include the global variable "iris"

File: Homework_1.py [hunk 8]
```diff
@@ -2,42 +2,109 @@
...
+        x="petal length",
+        y="petal width",
+        color="species",
+    )
+    bar_chart.show()
+
+
+# 4. standardscaler transformer using sk learn
+def standardization():
+    global x_scaled
```
Comment (cmalik3595) [inline Homework_1.py:79]: iris is a local variable in this def. Please add global keyword to include the global variable "iris"

File: Homework_1.py [hunk 9]
```diff
@@ -2,42 +2,109 @@
...
+    s.head()
+    scaler = StandardScaler().fit(s)
+    print(scaler)
+    x_scaled = scaler.transform(s)
+    print(x_scaled)
+
+
+# 5. fitting transformed data against random forest classifier
+def random_forest():
+    x = x_scaled[:, [0, 1, 2, 3]]
```
Comment (cmalik3595) [inline Homework_1.py:91]: iris is a local variable in this def. Please add global keyword to include the global variable "iris"

File: Homework_1.py [hunk 10]
```diff
@@ -2,42 +2,109 @@
...
+x_scaled = []
+x = []
+y = []
 
-n_array = df.to_numpy()
 
-print(n_array)
+def load_database():
+    global iris
+    iris = datasets.load_iris()
```
Comment (cmalik3595) [inline Homework_1.py]: load_iris returns a dict object, whereas iris is a dataframe object.

**Review by cmalik3595 — CHANGES_REQUESTED** [review]

File: Homework_1.py (hunk 6 above)
Comment (Anujagagangras) [reply Homework_1.py:40]: added iris as a global variable.

File: Homework_1.py (hunk 8 above)
Comment (Anujagagangras) [reply Homework_1.py:79]: added iris as a global variable.

File: Homework_1.py (hunk 10 above)
Comment (Anujagagangras) [reply Homework_1.py]: added iris as a global variable.

Comment (Anujagagangras) [reply Homework_1.py]: added iris_data as an object and converted to dataframe called iris.

File: Homework_1.py (hunk 9 above)
Comment (Anujagagangras) [reply Homework_1.py:91]: added iris as a global variable.

**Review by cmalik3595 — APPROVED** [review]
Thanks. The code looks okay to me. Please make sure it runs correctly and give appropriate results. Thanks.

**Comment by dafrenchyman** [comment]
You shouldn't have merged this in yet... not until I grade it!

**Pull request merged**